### PR TITLE
Add file comments for source modules

### PR DESCRIPTION
**Pull request by palash25**

### CODE REVIEW

**Comment by palash25** [comment]
@bieniusa can you PTAL?

**Comment by bieniusa** [comment]
Thank you for your effort! 

I find your descriptions to the point, but imho they are a bit short - is there enough information to help people navigate the code?

@albsch @peterzeller Should we merge them now and extend later?

**Comment by palash25** [comment]
I could try adding more information if you want @bieniusa .Can you give me an example on how much details would you like to see in the doc comments. it would be nice to have some kind of reference to be able to add  more details to the rest of the comments. :smile:

File: src/cure.erl [hunk 1]
```diff
@@ -26,6 +26,9 @@
 %% Description and complete License: see LICENSE file.
 %% -------------------------------------------------------------------
 
+%% @doc Provides a protocol to replicate updates from one cluster to another
+%%      in the form of a transaction API.
+
```
Comment (albsch) [inline src/cure.erl]: Maybe give a short description of CURE, the functions it provides, why it needs fixed static sharding, and reference the paper

File: src/dc_utilities.erl [hunk 2]
```diff
@@ -26,6 +26,8 @@
 %% Description and complete License: see LICENSE file.
 %% -------------------------------------------------------------------
 
+%% @doc Provides utility functions for data centers.
+
```
Comment (albsch) [inline src/dc_utilities.erl]: Add as a note that this module acts as an adapter for `riak_core`

File: src/inter_dc_manager.erl [hunk 3]
```diff
@@ -26,6 +26,8 @@
 %% Description and complete License: see LICENSE file.
 %% -------------------------------------------------------------------
 
+%% @doc Provides an API to interact nodes within a DC.
+
```
Comment (albsch) [inline src/inter_dc_manager.erl]: Short summary of what interactions are possible

File: src/inter_dc_query_response.erl [hunk 4]
```diff
@@ -26,6 +26,8 @@
 %% Description and complete License: see LICENSE file.
 %% -------------------------------------------------------------------
 
+%% @doc Starts a GenServer process and exposes an API to get node entries
+%%      and permissions.
```
Comment (albsch) [inline src/inter_dc_query_response.erl]: Short explanation of what node entries and permissions are

File: src/materializer.erl [hunk 5]
```diff
@@ -26,6 +26,9 @@
 %% Description and complete License: see LICENSE file.
 %% -------------------------------------------------------------------
 
+%% @doc Responsible for generating and caching the object versions
+%%      requested by clients.
+
```
Comment (albsch) [inline src/materializer.erl]: This module is only for generating the object versions, not for caching

File: src/materializer_vnode.erl [hunk 6]
```diff
@@ -26,6 +26,8 @@
 %% Description and complete License: see LICENSE file.
 %% -------------------------------------------------------------------
 
+%% @doc Provides an API to read and write to the materializer cache.
+
```
Comment (albsch) [inline src/materializer_vnode.erl]: Short explanation of why this module needs to be a `vnode`

**Comment by albsch** [comment]
As an example for how much doc should be at least at the top level doc, see the [bounded counter manager](https://github.com/AntidoteDB/antidote/blob/master/src/bcounter_mgr.erl) or the [clocksi coordinator](https://github.com/AntidoteDB/antidote/blob/master/src/clocksi_interactive_coord.erl). More documentation is better, of course.

File: src/inter_dc_query_response.erl (hunk 4 above)
Comment (palash25) [reply src/inter_dc_query_response.erl]: I couldn't find a lot of docs on this, especially permissions. Could you help me out here?

**Comment by palash25** [comment]
@albsch I have addressed all review comments except for one, please take a look again :smile:

File: src/cure.erl (hunk 1 above)
Comment (albsch) [reply src/cure.erl]: Still missing the fixed sharding bit

File: src/inter_dc_query_response.erl (hunk 4 above)
Comment (albsch) [reply src/inter_dc_query_response.erl]: There is no doc on this, as far as I know. `request_permissions` is a special inter dc request for the bounded counter manager.

File: src/materializer_vnode.erl (hunk 6 above)
Comment (albsch) [reply src/materializer_vnode.erl]: Here it's not important *what* a vnode is, but *why* a vnode is used.

**Comment by palash25** [comment]
@albsch can you take a look again? I couldn't really find anything about request permissions, I tried reading the code but even that didn't help in making me understand what they were, would you accept this as it is or should I remove that file comment entrirely?

**Comment by palash25** [comment]
thanks

**Comment by albsch** [comment]
Why did you remove the bits I added/changed?

**Comment by palash25** [comment]
did you amend my commit @albsch ? I didn't see any changes, previously my PR was failing the travis build because of a lint check, so I fixed it amended my commit and force pushed it.

**Comment by bieniusa** [comment]
Using force push seems to be a bad idea. Why would this be necessary here?

**Comment by albsch** [comment]
I edited it on Github on the fly and commited it. Now it's gone.

**Comment by albsch** [comment]
I'll redo my review soon.

**Comment by palash25** [comment]
I needed a force push to push my amended commit since I didn't want to make an unnecessary commit to fix a lint check @bieniusa

**Comment by palash25** [comment]
sorry for the inconvenience @albsch I didn't know you were going to edit it online

**Pull request merged**